File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    counts = Counter(tangerine)
    
    sorted_counts = sorted(counts.values(), reverse=True)
    
    answer = 0
    total = 0
    
    for count in sorted_counts:
        total += count
        answer += 1
        
        if total >= k :
            return answer        

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const counts = {};
    for (const size of tangerine) {
        counts[size] = (counts[size] || 0) + 1;
    }
    
    const sortedCounts = Object.values(counts).sort((a, b) => b - a);
    
    let answer = 0;
    let sum = 0;
    for (const count of sortedCounts) {
        sum += count;
        answer++;
        if (sum >= k) break;
    }
    
    return answer;
}